### PR TITLE
Test fixtures

### PR DESCRIPTION
**Pull request by xrotwang**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "clts"]
+    path = tests/fixtures/clts
+    url = https://github.com/cldf-clts/clts
```
Comment (xrotwang) [inline .gitmodules:3]: The CLTS data is needed for the regression test suite.

File: src/soundvectors.py [hunk 2]
```diff
@@ -9,6 +9,12 @@
 import collections
 import dataclasses
 
+try:  # pragma: no cover
+    from pyclts import CLTS, TranscriptionSystem
```
Comment (xrotwang) [inline src/soundvectors.py:13]: We only try to import from pyclts for type hinting.

File: src/soundvectors.py [hunk 3]
```diff
@@ -21,6 +27,16 @@ class Sound(typing.Protocol):  # Sound objects are expected to have a name attri
...
 
 
+class DomainHierarchyType(typing.Protocol):
+    @classmethod
+    def primary_domains(cls) -> typing.Set:
+        pass
+
+    @classmethod
+    def exclude_for_to_sound(cls) -> typing.Set:
+        pass
```
Comment (xrotwang) [inline src/soundvectors.py:37]: Here's the methods expected for a DomainHierarchy implementation - for type hints.

File: src/soundvectors.py [hunk 4]
```diff
@@ -706,7 +718,10 @@ def validate(self, sound):
             if is_valid_sound(sound.name):
                 return sound.name
         elif sound and self.ts:
-            sound_transcribed = self.ts([sound])[0]
+            if isinstance(self.ts, TranscriptionSystem):
+                sound_transcribed = self.ts[sound]
```
Comment (xrotwang) [inline src/soundvectors.py:722]: sound lookup the pyclts way.

File: tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+def pytest_generate_tests(metafunc):
+    def iter_sounds():
+        for line in FIXTURES.joinpath('sounds.tsv').read_text(encoding='utf-8').split('\n'):
+            if line:
+                sound, _, vec = line.partition('\t')
+                yield sound, eval(vec)
+
+    sounds = list(iter_sounds())
+    if not metafunc.config.getoption("all"):
+        sounds = random.sample(sounds, 200)
```
Comment (xrotwang) [inline tests/conftest.py:28]: Since all >8000 tests take about 25secs, I only sample 200 randomly for ordinary, in-development test runs.

**Comment by xrotwang** [comment]
@LinguList `cldf-clts/clts` as submodule for the tests works quite well - also in GH actions. So including tests that use `pyclts` shouldn't be a problem.

File: src/soundvectors.py (hunk 4 above)
Comment (arubehn) [reply src/soundvectors.py:723]: does this still work with `linse`?

Comment (xrotwang) [reply src/soundvectors.py:723]: The next lines provide compat with `linse`. The `ifinstance` makes sure this line is only called with `pyclts.TranscriptionSystem` instances.

Comment (arubehn) [reply src/soundvectors.py:723]: Ah, perfect. Just saw that, the else statement was not included in the conversation overview :)

**Review by arubehn — APPROVED** [review]

**Comment by LinguList** [comment]
Perfect! Also good for other project we might pursue in the future!

**Review by arubehn — APPROVED** [review]

**Comment by LinguList** [comment]
I suggest, we can submit this to PyPi any time when you are all fine. We should do before 10th of May, to be able to include the URL with the version in the paper draft.

**Comment by xrotwang** [comment]
fine with me. I'd recommend following the procedure described in https://github.com/cldf-clts/soundvectors/blob/main/RELEASING.md

**Pull request merged**